README.md

```diff
@@ -1,2 +1,193 @@
+
+1. Provide a name for the token and accept Asana's API terms checkbox after reading them.
+2. Click on the **Create token** button.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/2-generate-token.png alt="Generate new PAT">
+
+### Step 5: Generate the token
+
+1. Copy the generated token and keep it secure. You will need this token to authenticate the Asana connector.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/3-copy-token.png alt="Copy PAT">
+
+## Quickstart
+
+To use the `Asana` connector in your Ballerina application, modify the `.bal` file as follows:
+
+### Step 1: Import the connector
+
+Import the `ballerinax/asana` package into your Ballerina project.
+
+```ballerina
+import ballerinax/asana;
+```
+
+### Step 2: Instantiate a new connector
+
+Create an `asana:ConnectionConfig` with the obtained PAT (or OAuth2) credentials and initialize the connector with it.
+
+```ballerina
+asana:ConnectionConfig asanaConfig = {
+    auth: {
+        token: authToken
+    }
+};
+
+asana:Client asana = check new (asanaConfig);
+```
+
+### Step 3: Invoke the connector operation
+
+Now, utilize the available connector operations.
+
+#### Get all projects for the authenticated user
+
+```ballerina
+Inline_response_200_19 asanaProjects = check asana->/projects();
+```
+
+#### Create a new task in a project
+
+```ballerina
+asana:Tasks_body taskReq = {
+    data: {
+        name: "Email Marketing Campaign",
+        notes: "Create a new email marketing campaign for the upcoming product launch.",
+        workspace: "<workspaceId>",
+        projects: ["<projectId>"]
+    }
+};
+
+var createdTask = check asana->/tasks.post(taskReq);
+```
+
+## Examples
+
+The `Asana` connector offers practical examples illustrating its use in various scenarios.
+Explore these [examples](https://github.com/ballerina-platform/module-ballerinax-asana/tree/master/examples/), covering the following use cases:
+
+1. [Employee onboarding process automation](https://github.com/ballerina-platform/module-ballerinax-asana/tree/master/examples/employee-onboarding-process-automation) - Automate the onboarding process of new employees using Asana projects and tasks.
+2. [Team workload balancer](https://github.com/ballerina-platform/module-ballerinax-asana/tree/master/examples/team-workload-balancer) - Evaluate and balance the workload of a given team using Asana tasks and assignments.
+
+
+## Issues and projects
+
+The **Issues** and **Projects** tabs are disabled for this repository as this is part of the Ballerina library. To report bugs, request new features, start new discussions, view project boards, etc., visit the Ballerina library [parent repository](https://github.com/ballerina-platform/ballerina-library).
+
+This repository only contains the source code for the package.
+
+## Build from the source
+
+### Prerequisites
```

Comment:
I think we might need to add the instruction to include Github PAT and username as it would be need to retrieve the Ballerina gradle plugin. @ThisaruGuruge any thoughts ?

Reply:
@ayeshLK you seem to be correct. tried with a invalid PAT and the gradle failed to resolve the Ballerina plugin. Updated the docs with https://github.com/ballerina-platform/module-ballerinax-asana/pull/3/commits/13b948a0403f21636d507cfa0d01298ae40a6493